src/libraries/Microsoft.PowerFx.Core/Utils/TextFormatUtils.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.RegularExpressions;
+using Microsoft.PowerFx.Core.Errors;
+using Microsoft.PowerFx.Core.Localization;
+using System.Xml.Linq;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx.Core.Utils
+{
+    internal sealed class TextFormatUtils
+    {
+        static public DateTime BaseDateTime => new DateTime(1899, 12, 30, 0, 0, 0, 0);
+
+        public static bool IsValidFormatArg(string formatArg, out bool hasDateTimeFmt, out bool hasNumericFmt)
+        {
+            // Verify statically that the format string doesn't contain BOTH numeric and date/time
+            // format specifiers. If it does, that's an error according to Excel and our spec.
+
+            // But firstly skip any locale-prefix
+            if (formatArg.StartsWith("[$-"))
+            {
+                var end = formatArg.IndexOf(']', 3);
+                if (end > 0)
+                {
+                    formatArg = formatArg.Substring(end + 1);
+                }
+            }
+
+            hasDateTimeFmt = formatArg.IndexOfAny(new char[] { 'm', 'M', 'd', 'D', 'y', 'Y', 'h', 'H', 's', 'S', 'a', 'A', 'p', 'P' }) >= 0;
```

Comment:
>new char[] { 'm', 'M', 'd', 'D', 'y', 'Y', 'h', 'H', 's', 'S', 'a', 'A', 'p', 'P' } [](http://example.com/codeflow?start=50&length=83)

[minor] consider making this a static readonly field, so that this array is only created once. Same with the next one

Reply:
This code was moved from Text class so it can be re-used. I'm trying to keep it close to original as much as possible.